sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -822,7 +822,7 @@ class Column(protected[sql] val expr: Expression) extends Logging {
    * @group expr_ops
    * @since 1.3.0
    */
-  def cast(to: String): Column = cast(DataTypeParser.parse(to))
```

Comment:
DataTypeParser.parse does not support LongTypes. I don't think their functionality quite overlap, I'd propose to leave cast as it is and to use fromString everywhere else. Does that work for you?